testing/unit/test_faucetizer.py

```diff
@@ -89,16 +169,137 @@ def tearDown(self):
         self._faucetizer = None
         self._cleanup_config_files()
 
-    def test_faucetize_simple(self):
-        """Test normal faucetize behavior"""
+    def test_devices_learned_and_authenticated(self):
+        """devices with different combinations of static and dynamic info"""
         self._faucetizer.reload_structural_config()
-        self._faucetizer.flush_behavioral_config(force=True)
 
-        expected_behavioral_config = yaml.safe_load(self.FAUCET_BEHAVIORAL_CONFIG)
-        with open(self._temp_behavioral_config_file) as temp_behavioral_config_file:
-            faucetizer_behavioral_config = yaml.safe_load(temp_behavioral_config_file)
+        placements = [
+            # mocking static placements
+            ('02:00:00:00:00:01', {'switch': 't2sw1', 'port': 1, 'connected': True}, True),
+            ('02:00:00:00:00:02', {'switch': 't2sw1', 'port': 2, 'connected': True}, True),
+            # devices dynamically learned
+            ('02:00:00:00:00:01', {'switch': 't2sw2', 'port': 2, 'connected': True}, False),
+            ('02:00:00:00:00:03', {'switch': 't2sw2', 'port': 1, 'connected': True}, False),
+            # devices expired
+            ('02:00:00:00:00:01', {'switch': 't2sw2', 'port': 2, 'connected': False}, False),
+            ('02:00:00:00:00:03', {'switch': 't2sw2', 'port': 1, 'connected': False}, False)
+        ]
 
-        self.assertEqual(faucetizer_behavioral_config, expected_behavioral_config)
+        behaviors = [
+            # mocking static behaviors
+            ('02:00:00:00:00:01', {'segment': 'SEG_A', 'role': 'red'}, True),
+            ('02:00:00:00:00:03', {'segment': 'SEG_C', 'role': 'blue'}, True),
+            # devices authenticated
+            ('02:00:00:00:00:02', {'segment': 'SEG_B', 'role': 'green'}, False),
+            ('02:00:00:00:00:03', {'segment': 'SEG_A', 'role': 'yellow'}, False)
+        ]
+
+        # process static device info
+        self._process_device_placement(placements[0])
+        self._process_device_placement(placements[1])
+        self._process_device_behavior(behaviors[0])
+        self._process_device_behavior(behaviors[1])
+
+        # process dynamic device info
+        self._process_device_behavior(behaviors[2])
+        self._process_device_placement(placements[2])
+        self._process_device_placement(placements[3])
+        self._process_device_behavior(behaviors[3])
+
+        expected_behavioral_config_str = """
```

Comment:
move definition outside of function -- it's too disruptive to the programmatic flow.  Should be a class variable.

Reply:
Done